src/category_theory/preadditive/Mat.lean

```diff
@@ -0,0 +1,486 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.default
+import category_theory.preadditive.single_obj
+import category_theory.preadditive.additive_functor
+import category_theory.limits.shapes.biproducts
+import category_theory.Fintype
+import algebra.big_operators.basic
+import data.matrix.notation
+
+/-!
+# Matrices over a category.
+
+When `C` is a preadditive category, `Mat_ C` is the preadditive categoriy
+whose objects are finite tuples of objects in `C`, and
+whose morphisms are matrices of morphisms from `C`.
+
+There is a functor `Mat_.embedding : C ⥤ Mat_ C` sending morphisms to one-by-one matrices.
+
+`Mat_ C` has finite biproducts.
+
+## The additive envelope
+
+We show that this construction is the "additive envelope" of `C`,
+in the sense that any additive functor `F : C ⥤ D` to a category `D` with biproducts
+lifts to a functor `Mat_.lift F : Mat_ C ⥤ D`,
+Moreover, this functor is unique (up to natural isomorphisms) amongst functors `L : Mat_ C ⥤ D`
+such that `embedding C ⋙ L ≅ F`.
+(As we don't have 2-category theory, we can't explicitly state that `Mat_ C` is
+the initial object in the 2-category of categories under `C` which have biproducts.)
+
+As a consequence, when `C` already has finite biproducts we have `Mat_ C ≌ C`.
+
+## Matrices over a ring
+
+We also provide for convenience `Mat R`, where `R` is a ring, as a category with objects `ℕ`,
```

Comment:
This is now no longer correct, right? The objects are `Fintype`, I think.
This should then also be fixed in the main PR post that will become the commit message.